lib/at_cmd_parser/at_utils.h

```diff
@@ -242,6 +242,40 @@ static inline bool is_command(const char *str)
 	return false;
 }
 
+/**
+ * @brief Check if a string is a beginning of an AT CLAC response
+ *
+ * This function will check if the string is a CLAC response prefix.
+ * Valid prefixes: AT+ and AT%, except AT%X
+ *
+ * @param[in] str String to examine
+ *
+ * @retval true  If the string is a CLAC response
+ * @retval false Otherwise
+ */
+static inline bool is_clac(const char *str)
+{
+	if (strlen(str) < 3) {
+		return false;
+	}
+
+	if ((toupper(str[0]) != 'A') || (toupper(str[1]) != 'T')) {
+		/* Not an AT command */
+		return false;
+	}
+
+	if ((toupper(str[2]) != '+') && (toupper(str[2]) != '%')) {
+		/* Neither AT+ nor AT% */
+		return false;
+	}
+
+	if ((toupper(str[2]) == '%') && (toupper(str[3]) == 'X')) {
```

Comment:
`str` is only guaranteed to be at least 3 characters, so `str[3]` might be out-of-bounds, although perhaps not very likely.

Reply:
Fixed